core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -462,6 +463,26 @@ private LoadTableResponse loadInternal(
 
   @Override
   public Table loadTable(SessionContext context, TableIdentifier identifier) {
+    try {
+      return loadTableInternal(context, identifier);
+    } catch (NoSuchTableException original) {
```

Comment:
I don't think is the right way to fallback to loading the metadata table.  If you look at the current fallback logic metadata tables, we should attempt loading in that same path.